src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -420,9 +551,16 @@ def postprocess(self, model_outputs, decoder_kwargs: Optional[Dict] = None, retu
                 # This won't work with left padding (which doesn't exist right now)
                 right_n = total_n - right
                 items = items[:, left:right_n]
+            if "Whisper" in self.model.__class__.__name__ and return_timestamps:
+                # Whisper needs the stride data
+                items = [items, stride]
             final_items.append(items)
-        if stride and self.type == "seq2seq":
-            items = _find_longest_common_sequence(final_items, self.tokenizer)
+        if stride and self.type == "seq2seq" and not return_timestamps:
+            items = _fast_find_longest_common_sequence(final_items, self.tokenizer)
+        elif stride and self.type == "seq2seq" and return_timestamps:
```

Comment:
This shouldn't have anything to do with `stride`.

In general we need to think arguments as orthogonally as possible, meaning they should act on objects regardless of the settings of other parameters. It's not always cleanly possible, but when we can do it parameters become automatically orthogonal (meaning they behave nicely in all possible configurations). 

I think we can do better here (probably not fully orthogonal though, but better).

Reply:
I think it's resolved now. (This still depends on stride being present, but the other path without stride also works now.